randovania/layout/base/available_locations.py

```diff
@@ -37,38 +37,69 @@ def description(self) -> str:
         raise ValueError(f"Unknown value: {self}")
 
 
+class LocationPickupMode(BitPackEnum, Enum):
+    SHUFFLED = "shuffled"
+    SHUFFLED_NO_PROGRESSION = "shuffled_no_progression"
+    SHUFFLED_NO_MAJORS = "shuffled_no_majors"
+
+    @property
+    def long_name(self) -> str:
+        if self == LocationPickupMode.SHUFFLED:
+            return "Shuffled"
+        elif self == LocationPickupMode.SHUFFLED_NO_PROGRESSION:
+            return "Shuffled (no progression)"
+        elif self == LocationPickupMode.SHUFFLED_NO_MAJORS:
+            return "Shuffled (no majors)"
+        else:
+            raise ValueError(f"Unknown: {self}")
+
+
 @dataclasses.dataclass(frozen=True)
 class AvailableLocationsConfiguration(BitPackValue):
     randomization_mode: RandomizationMode
     excluded_indices: frozenset[PickupIndex]
+    minor_only_indices: frozenset[PickupIndex]
     game: RandovaniaGame
 
     @property
-    def _sorted_indices(self) -> list[int]:
+    def _sorted_excluded_indices(self) -> list[int]:
         return list(sorted(item.index for item in self.excluded_indices))
 
+    @property
+    def _sorted_minor_only_indices(self) -> list[int]:
+        return list(sorted(item.index for item in self.minor_only_indices))
+
     @property
     def as_json(self) -> dict:
         return {
             "randomization_mode": self.randomization_mode.value,
-            "excluded_indices": self._sorted_indices,
+            "excluded_indices": self._sorted_excluded_indices,
+            "minor_only_indices": self._sorted_minor_only_indices,
         }
 
     @classmethod
     def from_json(cls, value: dict, game: RandovaniaGame) -> "AvailableLocationsConfiguration":
         return cls(
             randomization_mode=RandomizationMode(value["randomization_mode"]),
             excluded_indices=frozenset(PickupIndex(item) for item in value["excluded_indices"]),
+            minor_only_indices=frozenset(PickupIndex(item) for item in value.get("minor_only_indices", [])),
```

Comment:
this is slightly irresponsible. we shouldn't be using `get()` in this context - should instead write a preset migration and guarantee the key will be present